File: 0045-jump-game-ii/0045-jump-game-ii.rb
# @param {Integer[]} nums
# @return {Integer}
def jump(nums)
    ans = l = r = ub = 0
    while r < nums.length-1
        (l..r).each{ |i| ub = [ub, i+nums[i]].max }
        l = r+1
        r = ub
        ans += 1
    end
    return ans
end

File: 0045-jump-game-ii/0045-jump-game-ii.go
func jump(nums []int) int {
  ans, l, r, ub := 0, 0, 0, 0
  for r < len(nums)-1 {
    for i:=l;i<=r;i++ {
      ub = max(ub, i+nums[i])
    }
    l, r, ans = r+1, ub, ans+1
  }
  return ans
}